src/character.cpp

```diff
@@ -1946,8 +1947,13 @@ void Character::on_dodge( Creature *source, float difficulty )
 
     // Even if we are not to train still call practice to prevent skill rust
     difficulty = std::max( difficulty, 0.0f );
-    practice( skill_dodge, difficulty * 2, difficulty );
 
+    // If training_level is set, treat that as the difficulty instead
+    if( training_level != 0.0 ) {
+        difficulty = training_level;
```

Comment:
It is not possible for this conditional to ever be fulfilled, nothing ever passes a float argument for training_level into these functions

Reply:
I believe I've fixed this. It was working in testing but I definitely had 0 and not 0.0f in a few places.